include/pybind11/typing.h

```diff
@@ -78,6 +78,29 @@ class Optional : public object {
     using object::object;
 };
 
+#if defined(__cpp_nontype_template_parameter_class)
+template <size_t N>
+struct StringLiteral {
+    // This struct is different than detail::descr since it accepts a string rather than a type.
```

Comment:
Oh, wow, got it, thanks. — I really was off track here. Looking again more carefully: Sorry I overlooked the `constexpr` (line below) before. I would have asked a different question (or no question) otherwise. I think it's best if you delete this comment again.